pkg/util/utils_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"testing"
+
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func Test_PortNumberPtr(t *testing.T) {
+	var exportedPort65535 gatewayv1a2.PortNumber = 65535
+	var exportedPort1 gatewayv1a2.PortNumber = 1
+	var exportedPort0 gatewayv1a2.PortNumber = 0
+	var exportedPort65536 gatewayv1a2.PortNumber = 65536
+
+	portNumberPtrTests := []struct {
+		name         string
+		port         int
+		expectedPort *gatewayv1a2.PortNumber
+	}{
+		{
+			name:         "invalid port number",
+			port:         0,
+			expectedPort: &exportedPort0,
+		},
+		{
+			name:         "valid port number",
+			port:         65535,
+			expectedPort: &exportedPort65535,
+		},
+		{
+			name:         "invalid port number",
+			port:         65536,
+			expectedPort: &exportedPort65536,
+		},
+		{
+			name:         "valid port number",
+			port:         1,
+			expectedPort: &exportedPort1,
+		},
+	}
+
+	for _, tc := range portNumberPtrTests {
+
```

Comment:
Nit: This line can probably be removed.

Reply:
9b7324e3df241bd455b0dcc4d1ab6794b0186c83